artemis/modules/domain_expiration_scanner.py

```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/env python3
+import datetime
+from typing import Any, Dict
+
+from karton.core import Task
+from whois import query  # type: ignore
+
+from artemis.binds import TaskStatus, TaskType
+from artemis.config import Config
+from artemis.domains import is_main_domain
+from artemis.module_base import ArtemisBase
+
+
+class DomainExpirationScanner(ArtemisBase):
+    """
+    Alerts if domain expiration date is coming.
+    """
+
+    identity = "domain_expiration_scanner"
+    filters = [{"type": TaskType.DOMAIN.value}]
+
+    def run(self, current_task: Task) -> None:
+        domain = current_task.get_payload(TaskType.DOMAIN)
+        if is_main_domain(domain):
+            now = datetime.datetime.now()
+            result: Dict[str, Any] = {}
+            domain_data = query(domain)
+            expiry_date = domain_data.expiration_date
+            days_to_expire = None
+            if expiry_date:
+                days_to_expire = (expiry_date - now).days
+            result["expiry_date"] = expiry_date
+            if (
+                days_to_expire
+                and days_to_expire <= Config.Modules.DomainExpirationScanner.DOMAIN_EXPIRATION_ALERT_IN_DAYS
+            ):
+                result["close_expiry_date"] = True
+                result["days_to_expire"] = days_to_expire
+                status = TaskStatus.INTERESTING
+                status_reason = (
+                    f"Scanned domain will expire in {days_to_expire} days."
+                    if days_to_expire != 1
+                    else f"Scanned domain will expire in {days_to_expire} day."
+                )
+            else:
+                status = TaskStatus.OK
```

Comment:
save TaskStatus.OK always if you don't save INTERESTING. Currently you save this status only if `is_main_domain` returns true.